### PR TITLE
Group 5

### PR DESCRIPTION
**Pull request by johanneskross**

### CODE REVIEW

File: src/main/java/Airport.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        List<MilitaryPlane> transportMilitaryPlanes = new ArrayList<>();
+        Plane maxPlane = null;
+
+        // print banner
+        System.out.println ("*********************************************");
+        System.out.println ("***** Largest Transport Military Plane ******");
+        System.out.println ("*********************************************");
+
+        for (Plane plane : planes) {
+            if (plane instanceof MilitaryPlane) {
```
Comment (jboettcher) [inline src/main/java/Airport.java:22]: don't use instanceof -> use polymorphism

File: src/main/java/Airport.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                if (((MilitaryPlane) plane).getType() == MilitaryType.TRANSPORT) {
+                    if (maxPlane != null && maxPlane.getMaxLoadCapacity() < plane.getMaxLoadCapacity()) {
+                        maxPlane = plane;
+                    }
+                }
+            } //if
+            else {
+
+            } // else
+        } //for
```
Comment (jboettcher) [inline src/main/java/Airport.java:32]: remove comment

File: src/main/java/Airport.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        for (Plane plane : planes) {
+            if (plane instanceof MilitaryPlane) {
+                if (((MilitaryPlane) plane).getType() == MilitaryType.TRANSPORT) {
+                    if (maxPlane != null && maxPlane.getMaxLoadCapacity() < plane.getMaxLoadCapacity()) {
+                        maxPlane = plane;
+                    }
+                }
+            } //if
+            else {
```
Comment (TabeaSchmidt) [inline src/main/java/Airport.java:29]: empty else

File: src/main/java/Airport.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        System.out.println ("*********************************************");
+
+        for (Plane plane : planes) {
+            if (plane instanceof MilitaryPlane) {
+                if (((MilitaryPlane) plane).getType() == MilitaryType.TRANSPORT) {
+                    if (maxPlane != null && maxPlane.getMaxLoadCapacity() < plane.getMaxLoadCapacity()) {
+                        maxPlane = plane;
+                    }
+                }
+            } //if
```
Comment (jboettcher) [inline src/main/java/Airport.java:28]: remove comment

File: src/main/java/Airport.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+public class Airport {
+
+    /**
+     * Prints transport military planes
+     */
+    private void printMaxTransportMilitaryPlane() {
+        List<MilitaryPlane> transportMilitaryPlanes = new ArrayList<>();
+        Plane maxPlane = null;
+
+        // print banner
```
Comment (jboettcher) [inline src/main/java/Airport.java:16]: create method printBanner()

File: src/main/java/Airport.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.*;
+
+public class Airport {
+
+    /**
+     * Prints transport military planes
+     */
+    private void printMaxTransportMilitaryPlane() {
+        List<MilitaryPlane> transportMilitaryPlanes = new ArrayList<>();
+        Plane maxPlane = null;
```
Comment (jboettcher) [inline src/main/java/Airport.java:14]: maxPlane --> maxLoadCapacityPlane

File: src/main/java/Airport.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+        // print banner
+        System.out.println ("*********************************************");
+        System.out.println ("***** Largest Transport Military Plane ******");
+        System.out.println ("*********************************************");
+
+        for (Plane plane : planes) {
+            if (plane instanceof MilitaryPlane) {
+                if (((MilitaryPlane) plane).getType() == MilitaryType.TRANSPORT) {
+                    if (maxPlane != null && maxPlane.getMaxLoadCapacity() < plane.getMaxLoadCapacity()) {
```
Comment (TabeaSchmidt) [inline src/main/java/Airport.java:24]: divide into two if-clauses for better testability

File: src/main/java/Airport.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        Plane maxPlane = null;
+
+        // print banner
+        System.out.println ("*********************************************");
+        System.out.println ("***** Largest Transport Military Plane ******");
+        System.out.println ("*********************************************");
+
+        for (Plane plane : planes) {
+            if (plane instanceof MilitaryPlane) {
+                if (((MilitaryPlane) plane).getType() == MilitaryType.TRANSPORT) {
```
Comment (jboettcher) [inline src/main/java/Airport.java:23]: avoid cast

Comment (jboettcher) [reply src/main/java/Airport.java:23]: maybe add method `isTransport`

File: src/main/java/Airport.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        System.out.println("capacity:" + maxPlane.getMaxLoadCapacity());
+    }
+
+    //Constructor
+    public Airport(List<? extends Plane> planes) {
+        this.planes = planes;
+        this.printMaxTransportMilitaryPlane();
+    }
+
+    private List<? extends Plane> planes;
```
Comment (TabeaSchmidt) [inline src/main/java/Airport.java:43]: we would place class variables and constructor at the beginning of the class

File: src/main/java/Airport.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+            } // else
+        } //for
+        System.out.println("model:" + maxPlane.getModel());
+        System.out.println("capacity:" + maxPlane.getMaxLoadCapacity());
+    }
+
+    //Constructor
+    public Airport(List<? extends Plane> planes) {
+        this.planes = planes;
+        this.printMaxTransportMilitaryPlane();
```
Comment (jboettcher) [inline src/main/java/Airport.java:40]: should this be part of the constructor

Comment (jboettcher) [inline src/main/java/Airport.java:40]: should this be part of the constructor?

File: src/main/java/Airport.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
+import models.MilitaryType;
+import Planes.MilitaryPlane;
+import Planes.Plane;
+
+import java.util.*;
+
+public class Airport {
+
+    /**
```
Comment (TabeaSchmidt) [inline src/main/java/Airport.java:9]: unclear and incomplete comment for function

File: src/main/java/Airport.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+     * Prints transport military planes
+     */
+    private void printMaxTransportMilitaryPlane() {
+        List<MilitaryPlane> transportMilitaryPlanes = new ArrayList<>();
```
Comment (jboettcher) [inline src/main/java/Airport.java:13]: unused variable

File: src/main/java/Airport.java (hunk 7 above)
Comment (jboettcher) [inline src/main/java/Airport.java:24]: what if 2 or more planes have the same capacity?

File: src/main/java/Airport.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
+    //Constructor
+    public Airport(List<? extends Plane> planes) {
+        this.planes = planes;
+        this.printMaxTransportMilitaryPlane();
+    }
+
+    private List<? extends Plane> planes;
+
+    public List<? extends Plane> getPlanes() {
```
Comment (jboettcher) [inline src/main/java/Airport.java:45]: comment style different: Multi line comment on top of `printMaxTransportMilitaryPlane`, single line comment on top of constructor, no comments for the last methods

File: src/main/java/Airport.java (hunk 7 above)
Comment (jboettcher) [reply src/main/java/Airport.java:24]: maxPlane is always null. Logical bug

File: src/main/java/Airport.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
+import models.MilitaryType;
+import Planes.MilitaryPlane;
```
Comment (jboettcher) [inline src/main/java/Airport.java:2]: package name lower case

File: src/main/java/Airport.java [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+import Planes.Plane;
+
+import java.util.*;
```
Comment (jboettcher) [inline src/main/java/Airport.java:5]: only import List

**Pull request closed without merging**